dv/dpi/i2cdpi/i2c_hat_id.cc

```diff
@@ -11,22 +11,22 @@
 // Captured EEPROM ID header from Raspberry Pi Sense HAT; this device is not Read Only and it
 // supports Byte/Page Writes too. This static signature is used to initialise the memory.
 static const uint8_t id[] = {
- 0x52, 0x2D, 0x50, 0x69, 0x01, 0x00, 0x03, 0x00,
- 0xF0, 0x03, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, // R-Pi............
- 0x2D, 0x00, 0x00, 0x00, 0xE7, 0x58, 0xD1, 0x95,
- 0xF1, 0x56, 0x28, 0xAB, 0xCB, 0x4E, 0x99, 0x42, // -....X...V(..N.B
- 0xC7, 0x79, 0xD6, 0xA3, 0x01, 0x00, 0x01, 0x00,
- 0x0C, 0x09, 0x52, 0x61, 0x73, 0x70, 0x62, 0x65, // .y........Raspbe
- 0x72, 0x72, 0x79, 0x20, 0x50, 0x69, 0x53, 0x65,
- 0x6E, 0x73, 0x65, 0x20, 0x48, 0x41, 0x54, 0x7F, // rry PiSense HAT.
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // ................
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // ................
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // ................
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
- 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00  // ................
+  0x52, 0x2d, 0x50, 0x69, 0x01, 0x00, 0x03, 0x00,
+  0x7f, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,  // R-Pi............
+  0x2d, 0x00, 0x00, 0x00, 0x8f, 0xa4, 0xf3, 0x5e,
+  0x01, 0x26, 0x27, 0x97, 0x7e, 0x41, 0xd4, 0x83,  // -......^.&'.~A..
+  0x12, 0x4c, 0xa9, 0xaf, 0x01, 0x00, 0x02, 0x00,
+  0x0c, 0x09, 0x52, 0x61, 0x73, 0x70, 0x62, 0x65,  // .L........Raspbe
+  0x72, 0x72, 0x79, 0x20, 0x50, 0x69, 0x53, 0x65,
+  0x6e, 0x73, 0x65, 0x20, 0x48, 0x41, 0x54, 0x3c,  // rry PiSense HAT<
+  0xfc, 0x02, 0x00, 0x01, 0x00, 0x20, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  // ..... ..........
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,  // ................
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xed,
+  0x6e, 0x03, 0x00, 0x02, 0x00, 0x0e, 0x00, 0x00,  // ........n.......
+  0x00, 0x72, 0x70, 0x69, 0x2d, 0x73, 0x65, 0x6e,
+  0x73, 0x65, 0x2d, 0x76, 0x32, 0x37, 0x54, 0xff   // .rpi-sense-v27T.
```

Comment:
The previous dump was missing half of the data due to a bug in the I2C driver (which is fixed with https://github.com/lowRISC/cheriot-rtos/pull/8). This new dump is taken from the sense HAT I have and was obtained with the I2C on a raspberry-pi. Some bytes differ in the first half due to the different UUID and CRCs

Reply:
wow, good to have this fixed!

Reply:
The EEPROM test is failing in CI. Is this related to your rework?